src/wallet/rpcwallet.cpp

```diff
@@ -3671,9 +3672,8 @@ class DescribeWalletAddressVisitor : public boost::static_visitor<UniValue>
             // Always report the pubkey at the top level, so that `getnewaddress()['pubkey']` always works.
             if (subobj.exists("pubkey")) obj.pushKV("pubkey", subobj["pubkey"]);
             obj.pushKV("embedded", std::move(subobj));
-        } else if (which_type == TxoutType::MULTISIG) {
-            // Also report some information on multisig scripts (which do not have a corresponding address).
-            // TODO: abstract out the common functionality between this logic and ExtractDestinations.
+        } else if (which_type == TxoutType::MULTISIG && pwallet->chain().rpcEnableDeprecated("addresses")) {
```

Comment:
Just noticed this: please don't remove this. This isn't doing anything confusing (it's reporting actual pubkeys, not addresses), and isn't restricted to bare multisig. Reporting detailed information about properly wrapped multisig isn't something we should remove.

Reply:
Thanks for catching that - I reverted this change and updated the docs. re-squashed